src/scheduler/server_info.cpp

```diff
@@ -798,6 +803,16 @@ query_server_dyn_res(server_info *sinfo)
 				log_eventf(PBSEVENT_DEBUG2, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
 					"%s = %s (\"%s\")", conf.dynamic_res[i].command_line, res_to_str(res, RF_AVAIL), buf);
 
+			if (pid != 0) {
```

Comment:
Since we are checking pid again, shouldn't we set it to 0 if we wait for it above? That way, we don't go through this code again.

Reply:
See above -- I think I will just change it so that we always kill the child at the end, regardless of what happens.

Bhroam is right that we should only do that if pid is positive.